src/utils/attributes.ts

```diff
@@ -0,0 +1,68 @@
+import { createObserver } from "./observers";
+
+type AttributeObject = { [k: string]: any };
+
+/**
+ * Watches global attributes of a component.
+ *
+ * Derived from: https://gist.github.com/willmartian/b4dd6b57d71dd0438fb9e7c6f4048578
+ */
+export interface WatchedComponent {
+  /**
+   * The host element.
+   */
+  readonly el: HTMLElement;
+
+  /**
+   * The watched attributes object.
+   * Should be stateful.
+   * '@State() inheritedAttributes = {};'
+   */
+  inheritedAttributes: AttributeObject;
+
+  /**
+   * The MutationObserver to listen to inherited attributes.
+   */
+  inheritedAttributesObserver: MutationObserver;
+}
+
+/**
+ * Helper to set up listening for changes to global attributes.
+ *
+ * render() {
+ *   const lang = this.inheritedAttributes['lang'] ?? 'en';
+ *   return <div>My lang is {lang}</div>;
+ * }
+ */
+export function watchAttributes(component: WatchedComponent, attributes: string[]): void {
```

Comment:
To help keep this strictly for inherited attribute usage, should we add an attribute allowlist? As an extra step, we can also use some TS magic to avoid watching anything that matches a prop from a component.

Reply:
Yeah, that sounds good. Should we just start with `["lang"]`?

Reply:
Sounds like a good start.